### PR TITLE
RUMM-2972 [SR] Support `UISegmentedControl` elements in session replay

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

📦 With this PR we're adding support to `UISegmentedControl` elements in session replay.

segment-light

### How?

- added `UISegmentRecorder` with `UISegmentWireframesBuilder`
- added "Segments" fixture and snapshot tests

The slider is constructed with 1 to many wireframes:
- one _shape_ wireframe for drawing the background of entire control
- separate _text_ wireframes for drawing segment labels
- one more _shape_ wireframe to highlight selected segment.

If privacy mode `.maskAll` is enabled, we obfuscate text and do not build selection wireframe:

segment-mask

I also enhanced the way we deal with system colors by introducing `SystemColors` construct. It acts as a collection of system colors, which are UI mode dependant (return different values for light- and dark-mode on iOS 13+). 

Here, an example of segments in dark mode:

segment-dark

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: DatadogSessionReplay/SRSnapshotTests/SRSnapshotTests.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -549,7 +549,7 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/ncreated/Framing";
 			requirement = {
-				branch = framer;
+				branch = "ship-framer";
```
Comment (ncreated) [inline DatadogSessionReplay/SRSnapshotTests/SRSnapshotTests.xcodeproj/project.pbxproj:552]: Pointing to stable branch in https://github.com/ncreated/Framing - I added basic text alignment options in there, so we can preview `SRTextPosition` in snapshot tests.

File: DatadogSessionReplay/Sources/Recorder/Utilities/SystemColors.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            return UIColor.secondarySystemFill.cgColor
+        } else {
+            // Fallback to iOS 16.2 light mode color:
+            return UIColor(red: 120 / 255, green: 120 / 255, blue: 128 / 255, alpha: 1).cgColor
+        }
+    }
+
+    static var tintColor: CGColor {
+        if #available(iOS 15.0, *) {
+            return UIColor.tintColor.cgColor
```
Comment (maxep) [inline DatadogSessionReplay/Sources/Recorder/Utilities/SystemColors.swift:48]: Why 15? this one is available since iOS7

**Review by maxep — APPROVED** [review]
Nice, looks great!

File: DatadogSessionReplay/Sources/Recorder/Utilities/SystemColors.swift (hunk 2 above)
Comment (ncreated) [reply DatadogSessionReplay/Sources/Recorder/Utilities/SystemColors.swift:48]: Indeed, the [Apple doc says iOS7](https://developer.apple.com/documentation/uikit/uiview/1622467-tintcolor), but this seems to not be true:

Screenshot 2023-02-21 at 15 20 37

Screenshot 2023-02-21 at 15 20 05

🤷

**Pull request merged**